test/wanda_web/controllers/execution_controller_test.exs

```diff
@@ -61,4 +63,104 @@ defmodule WandaWeb.ExecutionControllerTest do
       end
     end
   end
+
+  describe "start execution" do
+    test "should start an execution", %{conn: conn} do
+      execution_id = UUID.uuid4()
+      group_id = UUID.uuid4()
+
+      json =
+        conn
+        |> put_req_header("content-type", "application/json")
+        |> post("/api/checks/executions/start", %{
+          "execution_id" => execution_id,
+          "group_id" => group_id,
+          "targets" => [
+            %{
+              "agent_id" => UUID.uuid4(),
+              "checks" => ["expect_check"]
+            }
+          ],
+          "env" => %{
+            "provider" => "azure"
+          }
+        })
+        |> json_response(202)
+
+      api_spec = ApiSpec.spec()
+      assert_schema(json, "AcceptedExecution", api_spec)
+    end
+
+    test "should return an error when attempting to start an already running execution", %{
```

Comment:
This test actually goes out of context here. We don't really need to test this here, it is already tested in the `start_execution` function.
If you want to test the `error` path, you should mock the `execution_server_impl`

Reply:
Yep, adjusted, thanks! 44a9eec360b38b02a997c84b527266137f459b95

Reply:
I would rename the test to avoid using `already running`.
Something like:
`should return an error when the start execution operation fails`

Reply:
To me is about making it clear what we are testing :smile:, in this case that starting an execution fails because it was already started. 

I am also fine in rephrasing as you suggest :wink:

Reply:
Changed